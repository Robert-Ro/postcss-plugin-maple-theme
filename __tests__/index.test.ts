import postcss, { CssSyntaxError } from 'postcss'
import plugin from '../src'
import postcssNested from 'postcss-nested'
// eslint-disable-next-line @typescript-eslint/no-var-requires
const postcssNesting = require('postcss-nesting')

describe('postcss-plugin-maple-theme test cases', () => {
  it('should function color property be resolved right', async () => {
    const result = postcss([
      plugin({
        groups: {
          G01: ['C01', 'C02'],
        },
        colors: {
          C01: '#eee',
          C02: '#111',
        },
      }),
      postcssNested(),
      postcssNesting(),
    ])
      .process(
        `a{
          color: cc(G01);
      }`
      )
      .root.toResult().css
    expect(result).toMatchSnapshot()
  })
  it('should no group colors will throw CssSyntaxError', async () => {
    expect(() => {
      postcss([
        plugin({
          groups: {
            G01: ['C01', 'C02'],
          },
          colors: {
            C01: '#eee',
            C02: '#111',
          },
        }),
        postcssNested(),
      ])
        .process(
          `a{
          color: cc(G02);
      }`
        )
        .root.toResult().css
    }).toThrow(CssSyntaxError)
  })
  it('should no postcssNested imported will throw CssSyntaxError', async () => {
    expect(() => {
      postcss([
        plugin({
          groups: {
            G01: ['C01', 'C02'],
          },
          colors: {
            C01: '#eee',
            C02: '#111',
          },
        }),
      ])
        .process(
          `a{
          color: cc(G01);
      }`
        )
        .root.toResult().css
    }).toThrow(CssSyntaxError)
  })
  it('useCustomProperties color property should be resolved right', () => {
    const result = postcss([
      plugin({
        groups: {
          G01: ['C01', 'C02'],
        },
        colors: {
          C01: 'red',
          C02: '--read',
        },
        useCustomProperties: true,
      }),
      postcssNested(),
    ])
      .process(
        `a{
          color: cc(G01);
      }
      `
      )
      .root.toResult().css
    expect(result).toMatchSnapshot()
  })
  it('non function color property should not be transformed', () => {
    const result = postcss([
      plugin({
        groups: {
          G01: ['C01', 'C02'],
        },
        colors: {
          C01: '#eee',
          C02: '#111',
        },
      }),
      postcssNested(),
    ])
      .process(
        `a{
          color: #FFF;
      }`
      )
      .root.toResult().css
    expect(result).toMatchSnapshot()
  })
})
