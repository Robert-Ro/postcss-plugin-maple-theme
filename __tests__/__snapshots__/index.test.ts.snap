// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`postcss-plugin-maple-theme test cases non function color property should not be transformed 1`] = `
"a{
          color: #FFF;
      }"
`;

exports[`postcss-plugin-maple-theme test cases should function color property be resolved right 1`] = `
"a{
          color: #eee;
          @nest html[data-theme=\\"dark\\"] &{
          color: #111
          }
      }"
`;

exports[`postcss-plugin-maple-theme test cases useCustomProperties color property should be resolved right 1`] = `
"a{
          color: var(--C01);
      }
          html[data-theme=\\"dark\\"] a{
          color: var(--C02);
}
      "
`;
